webapp/src/modules/nft/sagas.spec.ts

```diff
@@ -353,39 +353,58 @@ describe('when handling the transfer NFT request action', () => {
       vendor = VendorFactory.build(nft.vendor)
       txHash = 'someHash'
     })
-    describe('and it has an rental with status OPEN', () => {
-      let rental: RentalListing
-      beforeEach(() => {
-        nft.openRentalId = 'aRentalId'
-        rental = {
-          id: nft.openRentalId,
-          status: RentalStatus.OPEN
-        } as RentalListing
+    describe('and the transaction finishes', () => {
+      describe('and it has an rental with status OPEN', () => {
+        let rental: RentalListing
+        beforeEach(() => {
+          nft.openRentalId = 'aRentalId'
+          rental = {
+            id: nft.openRentalId,
+            status: RentalStatus.OPEN
+          } as RentalListing
+        })
+        it('should dispatch an action signaling the success of the action handling and cancel an existing rental listing', () => {
+          return expectSaga(nftSaga)
+            .provide([
+              [call(VendorFactory.build, nft.vendor), vendor],
+              [select(getWallet), wallet],
+              [select(getRentalById, nft.openRentalId!), rental],
+              [
+                call([vendor.nftService, 'transfer'], wallet, address, nft),
+                Promise.resolve(txHash)
+              ],
+              [call(waitForTx, txHash), Promise.resolve()],
+              [
+                call(waitUntilRentalChangesStatus, nft, RentalStatus.CANCELLED),
+                Promise.resolve()
+              ]
+            ])
+            .put(transferNFTSuccess(nft, address, txHash))
```

Comment:
If I'm not wrong, we're missing the put assertion here, right?
```suggestion
            .put(transferNFTransactionSubmitted(nft, txHash))
            .put(transferNFTSuccess(nft, address, txHash))
```